### ISSUE TITLE
Global Styles: Allow themes to opt out of core theme.json

### ISSUE THREAD

**Issue by scruffian** [WordPress/gutenberg#40181]
The core theme.json (as well as potentially the styles in block.json: https://github.com/WordPress/gutenberg/pull/34180) represent some opinions which some themes may not want to inherit. We should build a mechanism in theme.json to disable them.

This is similar to what the `wp-block-styles` theme support does at the moment. See https://github.com/WordPress/gutenberg/issues/37255.

**Comment by oandregal** [comment]
Just wanted to share https://github.com/WordPress/gutenberg/issues/37550 as it touches on this specific thing (absorb the remaining CSS bits so themes can override easily). At some point, we may want to introduce a flag that allows to "reset" all core styles. Though, I wouldn't do that until we absorb the remaining pieces (block styles & block style variations).

**Comment by scruffian** [comment]
Still relevant

**Comment by oandregal** [comment]
Looked at the related conversations, and I'm having trouble understanding what this means.

This is what I understand:

- We are trying to migrate the block styles to the global styles system (via the `__experimentalStyles` in `block.json` so far).
- When doing so, we need to build it in a way that respects the [wp-block-styles](https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-support/#default-block-styles) theme support. This is, some block styles are always enqueued and some other (more opinionated) styles won't unless the theme opts-in. We need to make sure to consider this when migrating opinionated styles to the `__experimentalStyles` approach.

Does that captures the goal?

This leaves me with a few questions:

- if the `theme.json` of the theme loads after core and blocks styles it can override anything, do we need to do anything else? Do we need perhaps a "reset" theme.json (ala "reset" CSS) for some scenarios?
- how the opt-in works with themes with and without `theme.json`?

In essence, my past self may have understood what this issue tries to capture, but my current self has some lacunas :) Could you offer some clarifications?

**Comment by scruffian** [comment]
I think it's much simper than that. WordPress itself contains a [base theme.json file](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/theme.json) which contains some preset colors, settings, styles etc. Themes should have a way to opt out of that entirely and start with a base that is empty.

**Comment by oandregal** [comment]
We already have a mechanism to "opt-out" or tweak core defaults. It's per datum (setting, style, etc). Perhaps my question can be boiled down to: what are the things themes can't do now by overriding the specific things from the core `theme.json`?

(I want to understand the needs before we jump into adding more complexity. See related https://github.com/WordPress/gutenberg/issues/45171 )

**Comment by scruffian** [comment]
We have a mechanism for themes to opt out of opinionated styles for blocks, using `wp-block-styles`. The suggestion here is to build the same affordance for theme.json. It is true that themes can override each of the default settings with their own, but we should also just allow themes to totally opt out of those defaults so they have a clean starting point.

**Comment by samikeijonen** [comment]
> but we should also just allow themes to totally opt out of those defaults so they have a clean starting point.

Yes, exactly this. Because in custom client work Core generated styles are pretty much always in the way and breaks existing styles when updating Core.

**Comment by tresorama** [comment]
Any news on this ?
